### PR TITLE
Set quota on target directory based on source actual size, not by its current consumed size

### PR DESCRIPTION
**Pull request by santhoshatdell**
# Description

- Set quota on target directory based on source actual size, not by its current consumed size.
- Updated version from 2.6.0 to 2.6.1

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/753 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes (79.9%)
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- Tested 10 PVs creation with no quotas and 20 PVs (5+5+10) with quotas enabled. When quotas is enabled, same quota gets set on both source and target directories.
- Tested RG sync and failover actions. Verified data writes to source before failover and writes to target after failover.
[Captured log snippets.txt](https://github.com/dell/csi-powerscale/files/11115405/Captured.log.snippets.txt)

Another round of test logs for QuotaScan workaround fix.
[Test log - SyncIQ - QuotaScan.txt](https://github.com/dell/csi-powerscale/files/11204091/Test.log.-.SyncIQ.-.QuotaScan.txt)

### CODE REVIEW

**Review by ChristianAtDell — APPROVED** [review]

**Review by anandrajak1 — APPROVED** [review]

**Comment by prablr79** [comment]
@santhoshatdell can we get the checklist fully checked for this PR ?

**Comment by santhoshatdell** [comment]
> @santhoshatdell can we get the checklist fully checked for this PR ?

Hi Prasanna,
I only work on replication related changes in this repo. When I do, I update replication related unit tests and try to keep up code coverage %. I cannot get the coverage to above 90%. We also have our e2e tests which we run to verify our fixes.

File: service/replication.go [hunk 1]
```diff
@@ -112,8 +112,16 @@ func (s *service) CreateRemoteVolume(ctx context.Context,
 	if err != nil {
 		return nil, status.Errorf(codes.Internal, "failed to sync data %s", err.Error())
 	}
-	volumeSize := isiConfig.isiSvc.GetVolumeSize(ctx, isiPath, volName)
-	log.Info("Volume size got: ", volumeSize)
+	// Get source quota size if assigned and set the same on target
+	var volumeSize int64
+	sourceQuota, err := isiConfig.isiSvc.client.GetQuotaWithPath(ctx, exportPath)
+	if err != nil && !strings.Contains(err.Error(), "not found:") {
```
Comment (donatwork) [inline service/replication.go:118]: Is the error coming from the array API or our code? Seems fragile if the API returns a different error. Hopefully they do not change this often.

**Review by donatwork — APPROVED** [review]

File: service/replication.go (hunk 1 above)
Comment (santhoshatdell) [reply service/replication.go:118]: This error is from our code in gopowerscale library. Array API just returns empty list if no quota is set for that path.

**Review by ChristianAtDell — APPROVED** [review]

**Pull request merged**